### PR TITLE
Stricten parsing of malformed parameter csv's with pandas>=1.4

### LINKED ISSUE #411: Relax pandas old version requirement
This should be relaxed:

https://github.com/equinor/ecl2df/blob/1cd9d887c75fb171100d2bbc70ad2cb02f9ddbdd/setup.py#L22

### PR DESCRIPTION
**Pull request by asnyv**
Resolves #411

### CODE REVIEW

**Comment by asnyv** [comment]
Used `index_col=False` to avoid the `pandas>1.4` behavior where 3 columns in the data and 2 defined `names` results in the first column being used as index. 
Then catching the warning `pandas` writes when some data is dropped (column 3) and raise it as an error instead.

**Comment by asnyv** [comment]
also closes #399

File: ecl2df/parameters.py [hunk 1]
```diff
@@ -65,9 +66,20 @@ def load_parameterstxt(filename: Union[str, Path]) -> Dict[str, Any]:
         filename: file containing one key-value pair pr. line,
             separated by whitespace
     """
-    dframe = pd.read_csv(
-        filename, comment="#", sep=r"\s", engine="python", names=["KEY", "VALUE"]
-    )
+    with warnings.catch_warnings(record=True):
+        warnings.filterwarnings("error")
```
Comment (berland) [inline ecl2df/parameters.py:70]: Maybe a small comment here is warranted for the future developer?

Comment (asnyv) [reply ecl2df/parameters.py:73]: done

**Review by berland — APPROVED** [review]
👍

**Pull request merged**